base/applications/regedit/txtproc.c

```diff
@@ -0,0 +1,402 @@
+}
+
+static void txt_export_multi_sz(FILE *fp, const void *data, size_t size)
+{
+    LPCWSTR pch;
+    for (pch = data; *pch; pch += lstrlenW(pch) + 1)
+    {
+        if (pch == data)
+            txt_export_field(fp, load_str(IDS_FIELD_DATA), pch);
+        else
+            txt_export_field(fp, L"", pch);
+    }
+}
+
+static void txt_export_type(FILE *fp, LPCWSTR type)
+{
+    txt_export_field(fp, load_str(IDS_FIELD_TYPE), type);
+}
+
+static void txt_export_name(FILE *fp, LPCWSTR name)
+{
+    txt_export_field(fp, load_str(IDS_FIELD_NAME), name);
+}
+
+static void
+txt_export_data(FILE *fp, INT i, LPCWSTR value_name, DWORD value_len, DWORD type,
+                const void *data, size_t size)
+{
+    LPCWSTR pszType;
+
+    txt_fprintf(fp, load_str(IDS_VALUE_INDEX), i);
+    txt_newline(fp);
+    txt_export_name(fp, value_name);
+
+    switch (type)
+    {
+        case REG_SZ:
+            txt_export_type(fp, L"REG_SZ");
+            txt_export_field(fp, load_str(IDS_FIELD_DATA), data);
+            break;
+
+        case REG_DWORD:
+            txt_export_type(fp, L"REG_DWORD");
+            txt_fprintf(fp, L"%-19s0x%lx\r\n", load_str(IDS_FIELD_DATA), *(DWORD*)data);
+            break;
+
+        case REG_EXPAND_SZ:
+            txt_export_type(fp, L"REG_EXPAND_SZ");
+            txt_export_field(fp, load_str(IDS_FIELD_DATA), data);
+            break;
+
+        case REG_MULTI_SZ:
+            txt_export_type(fp, L"REG_MULTI_SZ");
+            txt_export_multi_sz(fp, data, size);
+            break;
+
+        case REG_BINARY:
+        case REG_QWORD:
+        case REG_NONE:
+        default:
+            if (type == REG_BINARY)
+                pszType = L"REG_BINARY";
+            else if (type == REG_QWORD)
+                pszType = L"REG_QWORD";
+            else if (type == REG_NONE)
+                pszType = L"REG_NONE";
+            else
+                pszType = load_str(IDS_UNKNOWN);
+
+            txt_export_type(fp, pszType);
+            txt_export_field(fp, load_str(IDS_FIELD_DATA), L"");
+            txt_export_binary(fp, data, size);
+            break;
+    }
+
+    txt_newline(fp);
+}
+
+static WCHAR *
+txt_build_subkey_path(LPCWSTR path, DWORD path_len, LPCWSTR subkey_name, DWORD subkey_len)
+{
+    WCHAR *subkey_path;
+    subkey_path = malloc((path_len + subkey_len + 2) * sizeof(WCHAR));
+    swprintf(subkey_path, L"%s\\%s", path, subkey_name);
+    return subkey_path;
+}
+
+static void txt_export_key_name(FILE *fp, LPCWSTR name)
+{
+    txt_export_field(fp, load_str(IDS_FIELD_KEY_NAME), name);
+}
+
+static void txt_export_class_and_last_write(FILE *fp, HKEY key)
+{
+    WCHAR szClassName[MAX_PATH];
+    DWORD cchClassName = _countof(szClassName);
+    FILETIME ftLastWrite, ftLocal, ftNull = { 0 };
+    SYSTEMTIME stLastWrite;
+    WCHAR sz1[64], sz2[64];
+
+    RegQueryInfoKey(key, szClassName, &cchClassName, NULL, NULL, NULL, NULL, NULL, NULL, NULL,
+                    NULL, &ftLastWrite);
```

Comment:
Since most of the code uses WCHARs, use the -W (Unicode variant) of the function. Also are you really sure that this function won't ever fail?

Reply:
`W` is appended to the name. Fixed.